R/lint.R

```diff
@@ -28,11 +28,10 @@
 #' @return An object of class `c("lints", "list")`, each element of which is a `"list"` object.
 #'
 #' @examples
-#' \dontrun{
-#'   lint("some/file-name.R") # linting a file
-#'   lint("a = 123\n")        # linting inline-code
-#'   lint(text = "a = 123")   # linting inline-code
-#' }
+#' f <- withr::local_tempfile(lines = "a=1", fileext = "R")
```

Comment:
Hmm, I just remembered that `withr` is only a suggested dependency. So we shouldn't be running this example unless we check withr is installed. that makes things more complicated (this applies to other examples too)

Reply:
You are right! Forgot about that.

I can just `tempfile()` then, instead of `withr::local_tempfile()`.

Reply:
Turns out, we are using `{withr}` in many examples. So I have instead chosen the `#' @examplesIf` directive here and elsewhere.

Reply:
Why not `cat()` and `lint(text=)` everywhere?

Reply:
this example is specifically about showing the different ways of using lint()

several other places are examples for get_source_expressions()

I only saw one (trailing_blank_lines_linter()) that would be better suited by the text= approach l

Reply:
Oh, yes, sorry. How does `@examplesIf` render in the actual docs? As long as they show up, I'd say it's fine for those examples that really need a tempfile.

Reply:
> How does @examplesIf render in the actual docs? 

We are already using it [here](https://github.com/r-lib/lintr/blob/a32f2d23d07d734358924e456b10f3d375fc70b3/R/absolute_path_linter.R#L11), and you can see how this is rendered in the docs [here](https://lintr.r-lib.org/dev/reference/absolute_path_linter.html).

Reply:
And in the R 3.6 docs?

Reply:
You mean, what they would like in pkgdown website, PDF manual, or in RStudio IDE help page?

I can at least answer the first one: they will just show up as unevaluated lines of code.

For example, this will render 

```r
#' @examplesIf getRversion() > '5.0'
#' sd(cars$speed)
#' sd(cars$dist)
```

to this

<img width="414" alt="Screenshot 2022-10-19 at 20 50 24" src="https://user-images.githubusercontent.com/11330453/196778631-6c436b0e-05fa-4480-893a-95b6ceb66ff7.png">

Reply:
Fair enough. I was thinking about the interactive R help pages displayed in RStudio.